components/vision.py

```diff
@@ -13,33 +18,39 @@ class Vision:
     FIELD_LAYOUT = robotpy_apriltag.loadAprilTagLayoutField(
         robotpy_apriltag.AprilTagField.k2023ChargedUp
     )
+    FORWARD_CAMERA_TRANSFORM = Transform3d(
+        Translation3d(-0.35, 0.01, 0.11), Rotation3d.fromDegrees(0, 0, 175)
+    )
+    STD_DEV_CONSTANT = 5.0
+    ANGULAR_STD_DEV_CONSTANT = 3.0
+    ZERO_DIVISION_THRESHOLD = 1e-6
+    POSE_AMBIGUITY_FACTOR = 5.0
+    POSE_AMBIGUITY_THRESHOLD = 0.2
+
+    VELOCITY_SCALING_THRESHOLD = 0.5
+    VELOCITY_SCALING_FACTOR = 2
+
+    CONF_EXP_FILTER_ALPHA = 0.8
 
     field: wpilib.Field2d
+    enabled = tunable(True)
+    use_resection = tunable(False)
 
     def __init__(self) -> None:
         self.camera = PhotonCamera("forward_camera")
         self.has_targets = False
+        self.targets = []
         self.last_timestamp = 0
-        self.pose_estimator = RobotPoseEstimator(
-            self.FIELD_LAYOUT,
-            PoseStrategy.CLOSEST_TO_REFERENCE_POSE,
-            [
-                (
-                    self.camera,
-                    Transform3d(
-                        Translation3d(-0.35, 0.005, 0.26624),
-                        Rotation3d.fromDegrees(0, 0, 180),
-                    ),
-                )
-            ],
-        )
+        self.confidence_accs = [0.0] * 8
 
     def setup(self) -> None:
         self.field_pos_obj = self.field.getObject("vision_pose")
 
     def execute(self) -> None:
+        if not Vision.enabled:
```

Comment:
I don't believe this is correct — this would get the tunable descriptor rather than the value of the tunable.